test/fuzz/transformation_load_test.cpp

```diff
@@ -293,6 +293,176 @@ TEST(TransformationLoadTest, BasicTest) {
   ASSERT_TRUE(IsEqual(env, after_transformation, context.get()));
 }
 
+TEST(TransformationLoadTest, DemoTestCase) {
+  /* (will remove this after PR ready)
+  I want to edit 'TransformationLoadTest' to work for both OpLoad and
+  OpAtomicLoad
+  ---------------------------------------------------------------
+  OpLoad:
+  %res_type_id = opCode(OpLoad) %res_id %ptr_id MemOperand(literal)(optional)
+  ----------------------------------------------------------------
+  OpAtomicLoad:
+  %res_type_id = opCode(OpAtomicLoad) %res_id %ptr_id %scope_id
+  %semantic_id(mask - 32-bitIntegerScalar)
+
+  load through Pointer using give semantics.
+
+  - **ptr_id** for atomic load must be one of this:
+  Uniform when, StorageBufferm, PhysicalStorageBuffer,
+  Workgroup, CrossWorkgroup, Generic AtomicCounter, Image, Function.
+  Type of result type must be the same of type the pointer point to.
+
+  - %scope_id is the define the scope of execution, (32-bit integer scalar)
+  If labeled as a memory scope,
+  it specifies the distance of synchronization from the current invocation, what
+  this means?
+  Scopes:
+    -Everything executing on all the execution devices in the system.
+    CrossDevice = 0
+    -Everything executing on the device executing this invocation
+    Device = 1
+    -All invocations for the invoking workgroup.
+    Workgroup = 2
+    -All invocations in the currently executing subgroup.
+    Subgroup = 3
+    - There is also
+    QueueFamily(current queue family), QueueFamilyKHR, ShaderCallKHR
```

Comment:
These are only allowed if certain capabilities are used, so you could either (1) ignore them completely, of (2) allow them only if the module has the right capabilities.